be/src/exec/vectorized/json_parser.cpp

```diff
@@ -141,7 +147,22 @@ Status JsonArrayParser::advance() noexcept {
 }
 
 std::string JsonArrayParser::left_bytes_string(size_t sz) noexcept {
```

Comment:
You need to add some test cases

Reply:
Actually, there're unit tests for this method.
The crash is led by `return std::string(_doc.current_location()).substr(0, sz);`, which would access illegal memory when the `_doc` content is not terminated with `\0`. But the implement of std:;string would append `\0` to the end.